### ISSUE TITLE
require 'openssl' failure won't be detected properly under JRuby

### ISSUE THREAD

**Issue by kares** [rubygems/rubygems#1245]
due `e.message =~ / -- openssl$/`

``` ruby
begin
  require 'openssl'
rescue LoadError => e
  raise unless (e.respond_to?(:path) && e.path == 'openssl') ||
               e.message =~ / -- openssl$/
end
```

https://github.com/rubygems/rubygems/blob/master/lib/rubygems/security.rb#L14

I'm curious could this be done in a more (JRuby) compatible way or does it needs to stay and JRuby-OpenSSL should put in "hacks" such as https://github.com/jruby/jruby-openssl/pull/41 ... all JRuby currently does under _openssl.rb_ is a simple: `require 'jopenssl/load'`

**Comment by tenderlove** [comment]
That's really weird.  I wonder why we don't just remove the `raise` completely.  Presumably RubyGems can function without openssl.  I don't get it.

**Comment by djberg96** [comment]
Strange code. Looks to me like there's some intertwining of package.rb, security.rb and security/signer.rb that needs to be untwined.

**Comment by kares** [comment]
what's up with this ... is it there to stay than?

**Comment by djberg96** [comment]
@kares, it definitely could use some refactoring, but it might be a while before we get to it. As I mentioned earlier, there are those three files that need to be disentangled from each other.

**Comment by copiousfreetime** [comment]
It looks like this was put in as a fix for #481 by @indirect back in 2013, so might be good to ask him.

**Comment by bronzdoc** [comment]
We're going to close this because it is so old, but if this problem is still happening with the latest version of RubyGems please feel free to let us know with a new issue!
